File: Components/WinForms/CS/RichEditMainDemo/Modules/FloatingObjects.cs
using System;
using DevExpress.XtraRichEdit.Services;

namespace DevExpress.XtraRichEdit.Demos {
    public partial class FloatingObjectsModule : DevExpress.XtraRichEdit.Demos.TutorialControl {
        public FloatingObjectsModule() {
            InitializeComponent();
            OpenXmlLoadHelper.Load("FloatingObjects.docx", richEditControl);
            new RichEditDemoExceptionsHandler(richEditControl).Install();
        }

        public override RichEditControl PrintingRichEditControl { get { return richEditControl; } }

        protected override void DoShow() {
            base.DoShow();
            richEditControl.Focus();
            ribbonControl1.SelectedPage = homeRibbonPage1;
        }

    }
}

File: Components/WinForms/VB/RichEditMainDemo/Modules/FloatingObjects.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraRichEdit.Services

Namespace DevExpress.XtraRichEdit.Demos
	Partial Public Class FloatingObjectsModule
		Inherits DevExpress.XtraRichEdit.Demos.TutorialControl
		Public Sub New()
			InitializeComponent()
			OpenXmlLoadHelper.Load("FloatingObjects.docx", richEditControl)
			CType(New RichEditDemoExceptionsHandler(richEditControl), RichEditDemoExceptionsHandler).Install()
		End Sub

		Public Overrides ReadOnly Property PrintingRichEditControl() As RichEditControl
			Get
				Return richEditControl
			End Get
		End Property

		Protected Overrides Sub DoShow()
			MyBase.DoShow()
            richEditControl.Focus()
            ribbonControl1.SelectedPage = homeRibbonPage1
		End Sub

		Private Sub richEditControl_SelectionChanged(ByVal sender As Object, ByVal e As EventArgs) Handles richEditControl.SelectionChanged
            floatingPictureToolsRibbonPageCategory1.Visible = richEditControl.IsFloatingObjectSelected
		End Sub
	End Class
End Namespace
